directors/directors_test.go

```diff
@@ -0,0 +1,131 @@
+package directors_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/ONSdigital/florence/directors"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestDirectorPrefixTrimming(t *testing.T) {
+
+	Convey("Given a request to '/foo/bar'", t, func() {
+		request, _ := http.NewRequest("GET", "/foo/bar", nil)
+
+		Convey("When the Director is called with a prefix of '/foo'", func() {
+			directors.Director("/foo")(request)
+
+			Convey("Then the proxied request path should be '/bar'", func() {
+				So(request.URL.String(), ShouldEqual, "/bar")
+			})
+		})
+	})
+
+	Convey("Given a request to '/foo/bar'", t, func() {
+		request, _ := http.NewRequest("GET", "/foo/bar", nil)
+
+		Convey("When the Director is called with an empty string", func() {
+			directors.Director("")(request)
+
+			Convey("Then the proxied request path should be '/foo/bar'", func() {
+				So(request.URL.String(), ShouldEqual, "/foo/bar")
+			})
+		})
+	})
```

Comment:
I debated this, the only reason they are not is because the request object actually gets modified by the test so I need to create a new one. It felt odd to create two up front in the Given. I am not really sold on them being separate either though, I just couldn't think of a solution I really liked 😢

Open to any suggestions 😄

Reply:
Thanks for pointing out the execution order for goconvey. I was certainly overcomplicating that!

Looks way better now 😄